### DISCUSSION TITLE
How can I type $page.data more specifically?

### DISCUSSION THREAD

**Discussion by 205g0** [sveltejs/kit#8433] [Q&A]
`export let data: PageData` provides the specific type for a specific path, the respective `+page.server.ts` and its parent `+layout.server.ts`.  In contrast `$page.data` can only be typed via `app.d.ts` for the entire app/all paths.

Can I somehow get the former (specific and inferring) behavior for `$page.data` too?

**Comment by cdcarson** [comment]
I'm  pretty sure this can't happen automatically -- i.e. the  inferred shape of `$page.data` does not pick up  the shape defined in `./$types` relative to where you reference it.  You've got to use type guards, or just wing it.
